contrib/long_distance_matching/util.h

```diff
@@ -0,0 +1,27 @@
+#ifndef LDM_UTIL_H
```

Comment:
You might be interested in using `lib/common/mem.h` for an equivalent set of functions,
since, ultimately, this will be the module to use when integrating your work into `zstd`.